src/blob.js

```diff
@@ -0,0 +1,272 @@
+    assert(typeof options.name === 'string', 'The name of the blob, `options.name` must be specified.');
+
+    this.container = options.container;
+    this.blobService = this.container.blobService;
+    this.cacheContent = options.cacheContent || false;
+
+    this.name = options.name;
+    this.type = options.type;
+    this.contentType = options.contentType;
+    this.contentLanguage = options.contentLanguage;
+    this.contentDisposition = options.contentDisposition;
+    this.cacheControl = options.cacheControl;
+  }
+}
+
+class BlockBlob extends Blob {
+
+  constructor(options) {
+    options.type = 'BlockBlob';
+    super(options);
+  }
+
+  async create(content) {
+    let blobOptions = {
+      blobType: this.type,
+      contentType: this.contentType,
+      contentLanguage: this.contentLanguage,
+      contentDisposition: this.contentDisposition,
+      cacheControl: this.cacheControl,
+    };
+    try {
+      let result = await this.blobService.putBlob(this.container.name, this.name, blobOptions, content);
+      this.eTag = result.eTag;
+      this._cache(content);
+    } catch (error) {
+      rethrowDebug(`Failed to create the blob "${this.name}" with error: ${error}`, error);
+    }
+  }
+
+  /**
+   * Load the content of the blob
+   */
+  async load() {
+    // load the content only if the eTag of our local data doesn't match the copy on the server
+    let options = {};
+    let blob;
+    if (this.cacheContent) {
+      options.ifNoneMatch = this.eTag;
+    }
+    try {
+      let blob = await this.blobService.getBlob(this.container.name, this.name, options);
+
+      // update the properties
+      this.eTag = blob.eTag;
+      this.contentType = blob.contentType;
+      this.contentLanguage = blob.contentLanguage;
+      this.contentDisposition = blob.contentDisposition;
+      this.cacheControl = blob.cacheControl;
+
+      return blob.content;
+    } catch (error) {
+      if (error && error.statusCode === 304) {
+        return this.content;
+      }
+      rethrowDebug(`Failed to load the blob "${this.name}" with error: ${error}`, error);
+    }
+  }
+
+  async update(options, content) {
+    options = options || {};
+    options.blobType = this.type;
+
+    try {
+      let result = await this.blobService.putBlob(this.container.name, this.name, options, content);
+      this.eTag = result.eTag;
+      this._cache(content);
+    } catch (error) {
+      rethrowDebug(`Failed to update the blob "${this.name}" with error: ${error}`, error);
+    }
+  }
+}
+
+/**
+ * An instance of DataBlockBlob is a reference to a azure blob which contains a JSON.
+ * The content of the DataBlockBlob is validated against the schema stored at the container level.
+ */
+class DataBlockBlob extends BlockBlob {
+
+  constructor(options) {
+    options.contentType = 'application/json';
+    super(options);
+  }
+
+  _validateJSON(content) {
+    if (content && this.container.schemaId) {
```

Comment:
when would we want to validate JSON against a schema without a content body to validate against?